classes/class-wc-rest-connect-shipping-label-eligibility-controller.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( class_exists( 'WC_REST_Connect_Shipping_Label_Eligibility_Controller' ) ) {
+	return;
+}
+
+class WC_REST_Connect_Shipping_Label_Eligibility_Controller extends WC_REST_Connect_Base_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v1';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'connect/label';
+
+
+	/**
+	 * @var WC_Connect_Shipping_Label
+	 */
+	protected $shipping_label;
+
+	/**
+	 * @var WC_Connect_Payment_Methods_Store
+	 */
+	protected $payment_methods_store;
+
+
+	protected $has_only_tax_functionality;
+
+	public function __construct(
+		WC_Connect_API_Client $api_client,
+		WC_Connect_Service_Settings_Store $settings_store,
+		WC_Connect_Logger $logger,
+		WC_Connect_Shipping_Label $shipping_label,
+		WC_Connect_Payment_Methods_Store $payment_methods_store,
+		$has_only_tax_functionality
+	) {
+		parent::__construct( $api_client, $settings_store, $logger );
+		$this->shipping_label             = $shipping_label;
+		$this->payment_methods_store      = $payment_methods_store;
+		$this->has_only_tax_functionality = $has_only_tax_functionality;
+	}
+
+	/**
+	 * Register the routes for the shipping label eligibility.
+	 */
+	public function register_routes() {
+		// Accept request without order_id
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/creation_eligibility',
+			array(
+				array(
+					'methods'             => 'GET',
+					'callback'            => array( $this, 'get_creation_eligibility' ),
+					'args'                => array(
+						'order_id' => array(
+							'required' => false,
+							'type'     => 'integer',
+							'validate_callback' => 'is_numeric',
+						),
+					),
+					'permission_callback' => array( $this, 'check_permission' ),
+				),
+			)
+		);
+
+		// Accept request with order_id
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<order_id>\d+)/creation_eligibility',
+			array(
+				array(
+					'methods'             => 'GET',
+					'callback'            => array( $this, 'get_creation_eligibility' ),
+					'args'                => array(
+						'order_id' => array(
+							'required' => true,
+							'type'     => 'is_numeric',
```

Comment:
I think this should be `integer` instead of `is_numeric`

Reply:
Addressed in d0cbf253b73aa61b76d6febbd73c4f1caa9154a3